lib/open_food_network/orders_and_fulfillments_report/customer_totals_report.rb

```diff
@@ -190,7 +190,16 @@ def columns
       # rubocop:enable Metrics/PerceivedComplexity
 
       def line_item_includes
-        [{ variant: :product, order: [:bill_address, :shipments] }]
+        [{ variant: { product: :supplier },
+           order: [:bill_address, :ship_address, :order_cycle, :adjustments, :payments,
+                   :user, :distributor, shipments: { shipping_rates: :shipping_method }] }]
+      end
+
+      private
+
+      def shipping_method(line_items)
+        line_items.first.order.shipments.first.
+          andand.shipping_rates.andand.first.andand.shipping_method
```

Comment:
I think this change is incorrect, through OrderShipment you have order.shipping_method so this can stay:
line_items.first.order.shipping_method

I think that selecting shipping_rates.andand.first is wrong, we should use shipment.shipping_method that gets the  selected shipping_method

Reply:
If I change this back to `line_items.first.order.shipping_method` and check the generated queries again, there is a huge increase in queries to `shipping_rate` (one for every shipment in every order in the results). It looks like this:

![shipping_rates](https://user-images.githubusercontent.com/9029026/68201576-ec4cce80-ffb9-11e9-8907-43d11e4dc134.png)

We are essentially doing: `Spree::LineItem.includes(order: { shipments: { shipping_rates: :shipping_method } })` to get the shipping method eager-loaded correctly, and it seems that a consequence of that is that calling `line_item.shipping_method` directly triggers a new query ever time, whereas the slightly messy method above with the full chain (and lots of `#andand`) does not trigger new queries.

Reply:
ok, at least we need to make it correct

in spree Shipment.rb:
```
    def shipping_method
      selected_shipping_rate.try(:shipping_method) || shipping_rates.first.try(:shipping_method)
    end
```

shipping_rates.where(selected: true).first || shipping_rates.first.try(:shipping_method)

so, the alternative would be:
```
 def shipping_method(line_items)
      shipping_rates = line_items.first.order.shipments.first.andand.shipping_rates
      shipping_rates.andnad.where(selected: true).andand.first || shipping_rates.andand.first.try(:shipping_method)
end
```

but again the suggestion could be the same as for the orders method.
we can re-implement shipment_decorator#shipping_method and use the is_loaded?

I am sorry I am asking for so many changes!

Reply:
Can an order order have multiple shipping methods in OFN?

Reply:
Okay, I've adapted `Spree::Shipment#selected_shipping_rate` and rechecked the queries. LGTM. :+1:

Reply:
Hmmm... the build is broken now, and it indicates the shipping methods selected previously were correct, and the ones selected now are incorrect...

Reply:
I adapted it with:

```ruby
def selected_shipping_rate
      if shipping_rates.loaded?
        shipping_rates.find(&:selected)
      else
        shipping_rates.where(selected: true).first
      end
 end
```

...but a few specs that check for the correct selection of a `shipping_rate` fail. I think the original method works because it's part of a single query with a number of joins, so the `shipping_rate` that gets selected is automatically the one that's related to both the `shipment` and `shipping_method`. Does that make sense?

Reply:
I just realized that I forgot about this point...
I know this has been merged but I think this is a bug.

I dont think the shipping method is selected in any other way while fetching the DB, it's only selected in the code: shipment.shipping_method...
I think we need to verify this.

Reply:
@luisramos0 I'll look in release testing.
Can you give a little more indication of what I should specifically test? That shipping methods are correctly pulled into reports? Is this general or is there a specific edge case I am look at?

Reply:
this is specific to the customer_totals report.
to try to see a bug here we would need an order with multiple shipping methods and verify that the shipping method shown in the report is not the one selected by the user.
but I think the current code selects the first shipping method, so we need to select the second shipping method available for that order.

Reply:
Ok I'm confused. The UI never allows more than one shipping method. I know that the data model generally allows this but this is always limited in the UI.
I've certainly never seen a case of an order with >1 shipping method and have no idea how it would come about.

Reply:
yes, 1 order = 1 shipping method but when you checkout there is a list of shipping methods you can choose and all those shipping methods are copied to the order in the database.

Reply:
the one you select is marked in a field shipping_rate.selected

Reply:
Ok. Lots of reports use the shipping_method.first approach I believe. This is not a new thing is it?

Reply:
it is a new thing, I am pretty sure this is the only place where we are doing this.

Reply:
And you are right that this introduces a bug. And I think this needs to be fixed before the release.